x/ibc/23-commitment/types/merkle.go

```diff
@@ -128,31 +135,292 @@ func (MerkleProof) GetCommitmentType() exported.Type {
 }
 
 // VerifyMembership verifies the membership pf a merkle proof against the given root, path, and value.
-func (proof MerkleProof) VerifyMembership(root exported.Root, path exported.Path, value []byte) error {
+func (proof MerkleProof) VerifyMembership(specs []*ics23.ProofSpec, root exported.Root, path exported.Path, value []byte) error {
```

Comment:
Hmm... I see the issue, you cannot pass in the ics23.ProofSpec from the runtime unless you assert that the proofs are all ics23.CommitmentProofs.

However, I do not believe you need to do this here. I feel this was done in a non-breaking way by pulling in the iavl and tendermint specs in when decoding the `ics23:*` merkle.ProofOps. This will guarantee we are using the proper spec. The only thing we need to pass in here extra to assert total correctness is the types of the proof ops.

Thus, we may require `["ics23:iavl", "ics23:tendermint"]` as the two-step proofs for a sdk chain (substore and multistore). Passing those strings in would allow us to keep using `merkle.ProofOp` and still validate everything.